src/main/java/de/refactoringBot/rest/ConfigurationController.java

```diff
@@ -54,43 +55,22 @@
 	 * @param repoService
 	 * @return
 	 */
-	@RequestMapping(value = "/createConfig", method = RequestMethod.POST, produces = "application/json")
+	@RequestMapping(method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
 	@ApiOperation(value = "Create Git-Konfiguration")
-	public ResponseEntity<?> add(
-			@RequestParam(value = "repoService", required = true, defaultValue = "Github") String repoService,
-			@RequestParam(value = "repoName", required = true, defaultValue = "Bot-Playground") String repoName,
-			@RequestParam(value = "ownerName", required = true, defaultValue = "Refactoring-Bot") String repoOwner,
-			@RequestParam(value = "botUsername", required = true) String botUsername,
-			@RequestParam(value = "botPassword", required = true) String botPassword,
-			@RequestParam(value = "botEmail", required = true) String botEmail,
-			@RequestParam(value = "botToken", required = true) String botToken,
-			@RequestParam(value = "analysisService", required = false, defaultValue = "sonarqube") String analysisService,
-			@RequestParam(value = "analysisServiceProjectKey", required = false) String analysisServiceProjectKey,
-			@RequestParam(value = "maxAmountRequests", required = true, defaultValue = "5") Integer maxAmountRequests) {
-		// Check if repository already exists in another configuration
-		Optional<GitConfiguration> existsConfig = repo.getConfigByName(repoName, repoOwner);
-		// If it does
-		if (existsConfig.isPresent()) {
-			return new ResponseEntity<String>("There is already an configuration for this repository!",
-					HttpStatus.CONFLICT);
-		}
-
+	public ResponseEntity<Object> add(
+			@RequestBody GitConfigurationDTO newConfiguration) {
 		// Init database config
-		GitConfiguration savedConfig = null;
```

Comment:
Wenn man die Annotation 'RequestBody' statt die vielen Parameter verwendet ist das GitConfigurationDTO Objekt nicht nötig, genauso wie der Mapper. Man kann hier einfach das normale GitConfiguration-Objekt übergeben und es speichern. Ich verwende aktuell die Parameter weil wir keine eigene GUI haben. Mit dem Request Body kann man nicht so einfach Beispielswerte einfügen und es ist viel unübersichtlicher zum ausfüllen. Ob ein RequestBody oder viele RequestParameter ist Geschmackssache. Wenn eine richtige GUI existiert würde, würde man dort die Parameter haben und daraus ein Objekt zeugen und dieses dann im Body an die Schnittstelle versenden. Dann würde ich auf jeden Fall einen RequestBody verwenden.

EDIT: Was mit noch auffällt. Die "grabber.createConfigurationForRepo" ist entfernt. Da wurden eigentlich die Eingaben des Users gecheckt (Email, Token, etc.) und daraus wurde das Config-Objekt automatisch erstellt (mit den ganzen Links). Aktuell müsste man die ganzen Links selber im RequestBody eingeben. Da müsste man auch aufpassen. Für den RequestBody würde ich ein eigenes Objekt erstellen das als Input (Inhalt entpricht dann den aktuellen Parametern). Dann mit dem Objekt statt den Parametern arbeiten

Reply:
Ich stimme dir zu dass RequestParameter einige Vorteile haben wenn man vermehrt die Swagger UI verwendet.
Grade zum Testen ist aber auch eine JSON sehr komfortable weil man einfach Konfigurationen kopieren oder irgendwo in einem Dokument speichern muss und so nicht jedesmal alles neu eintragen muss. 
Ich persönlich würde deshalb den RequestBody drinlassen.

Reply:
Ja da gebe ich dir recht. Als Entwickler ist der RequestBody ganz praktisch da man seine JSON einfach reinkopieren kann.